packages/account/src/Sections/Profile/PersonalDetails/__tests__/verify-button.spec.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { VerifyButton } from '../verify-button';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import { Router } from 'react-router';
+import { createBrowserHistory } from 'history';
+import { routes } from '@deriv/shared';
+
+describe('VerifyButton', () => {
+    const history = createBrowserHistory();
+    const renderWithRouter = (component: React.ReactElement) => {
+        return render(<Router history={history}>{component}</Router>);
+    };
+
+    const mock_store = mockStore({
+        client: {
+            account_settings: {
+                phone_number_verification: {
+                    verified: false,
+                },
+            },
+        },
+    });
+
+    it('should render Verify Button', () => {
+        renderWithRouter(
+            <StoreProvider store={mock_store}>
+                <VerifyButton />
+            </StoreProvider>
+        );
+        expect(screen.getByText('Verify')).toBeInTheDocument();
+    });
+
+    it('should redirect user to phone-verification page when clicked on Verify Button', () => {
+        renderWithRouter(
+            <StoreProvider store={mock_store}>
+                <VerifyButton />
+            </StoreProvider>
+        );
+        const verifyButton = screen.getByText('Verify');
+        userEvent.click(verifyButton);
+        expect(history.location.pathname).toBe(routes.phone_verification);
+    });
+
+    it('should render Verified text', () => {
+        mock_store.client.account_settings.phone_number_verification.verified = true;
+        renderWithRouter(
+            <StoreProvider store={mock_store}>
+                <VerifyButton />
+            </StoreProvider>
+        );
+        expect(screen.getByText('Verified')).toBeInTheDocument();
+        expect(screen.getByTestId('dt_phone_verification_popover')).toBeInTheDocument();
```

Comment:
just a nitpick:
from the user experience perspective it would be more proper to test the popover with text visible on the screen, rather than testid...